src/components/header/NavCloseButton/NavCloseButton.tsx

```diff
@@ -15,9 +15,10 @@ export const NavCloseButton = ({
       className={classes}
       onClick={onClick}
       data-testid="navCloseButton"
+      aria-label="Close"
       {...buttonProps}
       type="button">
-      <Icon.Close size={3} aria-label="Close" />
```

Comment:
this is an acceptable implementation to me, my ANDI output (which shows the text data that is received by screen readers visually) shows the close button labelled as 'Close' as intended. 

<img width="374" alt="image" src="https://github.com/trussworks/react-uswds/assets/59394696/9d6e51d4-502a-4fdf-ba69-0bf082e5bc0f">


However, I would encourage you to get more specific in the accessible name. Let's say we had another component in a page layout that also had a close icon. If both are named "close" they would be indistinguishable for a screen reader user navigating by links on a page.

the aria label is correct but I would change it to "close menu" or "close navigation" to avoid duplicate accessible names. If there was two components using close buttons under this pattern in the same page layout, we would see this ANDI error: [⚠ Non-unique button: same name/description as another button](https://www.ssa.gov/accessibility/andi/help/alerts.html?non_unique_button).

Reply:
What about enabling the prop to be passed in? 

Whatever default value we provide here, whether "close" or "close menu" might be nice to have the option to override it. That would require an `iconProps` param passed in where we could then get `iconProps['aria-label']`, and if present, use that instead of the default.

I'm fine with that being an enhancement later on too though.

Reply:
@shkeating thanks for the review and the advice! I've updated the `aria-label` to `"Close Navigation Menu"` to make it more unique and descriptive.

@brandonlenz Allowing the label to be passed in as a prop sounds like the ideal path to me! I think this would be especially useful if we want to show localized aria labels.

I think there's a couple of other components that could use that enhancement as well, so I'll keep the diff on this PR as lean as possible for now and open a new PR targeted at using those props as overrides. Let me know if that works!

Reply:
That works for me.